### DISCUSSION TITLE
How to print BC/IC's dimension and the head data

### DISCUSSION THREAD

**Discussion by jacktang** [lululxvi/deepxde#718] [General]
Hello

Deepxde is highly depends on the BC/IC, and my question is how to debug BC/IC? say print the dimension of them and display the head data.

Thanks!

**Comment by jacktang** [comment]
I mean the error message like 
`ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 3 and the array at index 1 has size 2` 
or 
`ValueError: operands could not be broadcast together with shapes (3,) (2,)` 
is not explicit, and not so friendly for debugging

**Comment by lululxvi** [comment]
Not sure what you are trying to do. These errors are from TensorFlow/PyTorch.

**Comment by jacktang** [comment]
@lululxvi 我想在报错的时候能够打印出tensor的size或者shape或者里面的数据信息，这样可以方便倒查错误所在。
目前我看到训练只有`model.train(...)` 方法，不知道有没有什么`on_error`的回调函数来方面调试
